server/audit_math/suite.py

```diff
@@ -1,718 +0,0 @@
-
-    Inputs:
-        n - the sample size to test
-        alpha - the risk limit as a fraction
-        contest - the contest being audited
-        winner, loser - the winner-loser pair to evaluate the risk measurement over
-        bp_stratum: a stratum object containing the ballot polling stratum
-        cvr_stratum: a stratum object containing the ballot comparison stratum
-        n_ratio: the ratio of the size of the CVR stratum to both strata.
-    Outputs:
-        A Fisher's combined maximized p-value for the given sample size, contest,
-        winner-loser pair, and strata.
-    """
-    n1_original = cvr_stratum.sample_size
-    n2_original = bp_stratum.sample_size
-
-    o1_rate, o2_rate, u1_rate, u2_rate = 0.0, 0.0, 0.0, 0.0
-    # Assume o1, o2, u1, u2 rates will be the same as what we observed in sample
-    if n1_original != 0:
-        o1_rate = cvr_stratum.misstatements[(winner, loser)]["o1"] / n1_original
-        o2_rate = cvr_stratum.misstatements[(winner, loser)]["o2"] / n1_original
-        u1_rate = cvr_stratum.misstatements[(winner, loser)]["u1"] / n1_original
-        u2_rate = cvr_stratum.misstatements[(winner, loser)]["u2"] / n1_original
-
-    n1 = math.ceil(n_ratio * n)
-    n2 = int(n - n1)
-
-    if (n1 < n1_original) or (n2 < n2_original):
-        return 1.0
-
-    o1 = (
-        math.ceil(o1_rate * (n - n1_original))
-        + cvr_stratum.misstatements[(winner, loser)]["o1"]
-    )
-    o2 = (
-        math.ceil(o2_rate * (n - n1_original))
-        + cvr_stratum.misstatements[(winner, loser)]["o2"]
-    )
-    u1 = (
-        math.floor(u1_rate * (n - n1_original))
-        + cvr_stratum.misstatements[(winner, loser)]["u1"]
-    )
-    u2 = (
-        math.floor(u2_rate * (n - n1_original))
-        + cvr_stratum.misstatements[(winner, loser)]["u2"]
-    )
-
-    # Because this is a hypothetical sample, we create a
-    # corresponding hypothetical stratum
-    hyp_sample_size = n1
-
-    hyp_misstatements: MISSTATEMENTS = {
-        (winner, loser): {"o1": o1, "o2": o2, "u1": u1, "u2": u2,}
-    }
-
-    hyp_cvr_stratum = BallotComparisonStratum(
-        cvr_stratum.num_ballots,
-        cvr_stratum.vote_totals,
-        hyp_misstatements,
-        hyp_sample_size,
-    )
-
-    # Set up the no-CVR stratum, assuming the sample looks like the
-    # prior round
-    prev_sample_size = bp_stratum.sample_size
-    hyp_sample: BallotPollingStratum.SAMPLE_RESULTS = {"hyp_round": {}}
-
-    cumulative_sample = bravo.compute_cumulative_sample(bp_stratum.sample)
-
-    # Add fake ballots to the hypothetical sample:
-    if prev_sample_size == 0:
-        # If no ballots have been sampled, assume the sample is roughly
-        # the margin
-        hyp_sample["hyp_round"] = {
-            winner: min(
-                int(n2 * (bp_stratum.vote_totals[winner] / bp_stratum.num_ballots)),
-                bp_stratum.vote_totals[winner],
```

Comment:
why the `min` with `bp_stratum.vote_totals[winner]` - seemed like the original code just used the value on L519

Reply:
I suppose it probably doesn't matter, but I didn't want to be sampling more ballots than we had, since I don't know what that means.